src/vs/workbench/parts/terminal/electron-browser/windowsShellHelper.ts

```diff
@@ -80,22 +93,30 @@ export class WindowsShellHelper {
 				}
 				// Otherwise, we go up the tree to find the next valid deepest child of the root
 				this._childProcessIdStack.pop();
-				return this.refreshShellProcessTree(this._childProcessIdStack[this._childProcessIdStack.length - 1], null);
+				return this.refreshShellProcessTree(this._childProcessIdStack[this._childProcessIdStack.length - 1], null, false);
 			}
 			// We only go one level deep when checking for children of processes other then shells
-			if (SHELL_EXECUTABLES.indexOf(path.basename(result[0].executable)) === -1) {
+			const baseName = path.basename(result[0].executable);
+			if (SHELL_EXECUTABLES.indexOf(baseName) === -1) {
+				if (baseName === 'conhost.exe') {
+					// We're inside an external console; go up 2 levels and remove the console from the children
```

Comment:
Adding the example structure in the PR to the comment here would be good to better visualize the structure we're working with.